addons/website_sale/models/website_snippet_filter.py

```diff
@@ -21,7 +21,7 @@ def _prepare_values(self, **kwargs):
             return []
         hide_variants = False
         search_domain = kwargs.get('search_domain')
-        if self.filter_id and search_domain and 'hide_variants' in search_domain:
```

Comment:
Why are you removing this ?
I'm not sure it's the cause of the issue, but I can only see the alternative products snippet when I'm editing the page, not when simply browsing it.

Reply:
In order to have `hide_variants` in the context of the handlers, it needs to be in `self` before this line: https://github.com/odoo/odoo/blob/724502c55632c242245aa2a75eefc2c37edce34f/addons/website_sale/models/website_snippet_filter.py#L104

If removing `filter_id` is indeed causing the issue, I guess alternatives are passing the value via `kwargs` instead, or adding a `self = self.with_context(hide_variants=True)` before it (although [linters don't really like that kind of thing](https://docs.astral.sh/ruff/rules/self-or-cls-assignment/))